src/app/workspace/project/project-settings.component.ts

```diff
@@ -17,139 +17,70 @@
+import {selectProjectByWorkspace} from '../../core/store/projects/projects.state';
+import {isNullOrUndefined} from "util";
+import {Subscription} from 'rxjs/Subscription';
+import {ProjectModel} from '../../core/store/projects/project.model';
+import {selectAllUsers} from '../../core/store/users/users.state';
+import {ProjectsAction} from '../../core/store/projects/projects.action';
+import {selectAllCollections} from '../../core/store/collections/collections.state';
+import {CollectionsAction} from '../../core/store/collections/collections.action';
 
 @Component({
-  templateUrl: './project-settings.component.html',
-  styleUrls: ['./project-settings.component.scss']
+  templateUrl: './project-settings.component.html'
 })
 export class ProjectSettingsComponent implements OnInit {
 
-  public project: Project;
-  private organizationCode: string;
-  public projectCode: string;
-  private originalProjectCode: string;
-  public collectionsCount: number;
+  public userCount$: Observable<number>;
+  public collectionsCount$: Observable<number>;
+  public project: ProjectModel;
 
-  @ViewChild('projectDescription')
-  public projectDescription: ElementRef;
-  private originalProjectName: string;
+  private subscription = new Subscription();
 
   constructor(private i18n: I18n,
-              private projectService: ProjectService,
               private router: Router,
               private store: Store<AppState>,
-              private collectionService: CollectionService,
               private notificationService: NotificationService) {
   }
 
-  public ngOnInit(): void {
-    this.store.select(selectWorkspace).pipe(
-      filter(workspace => !!(workspace.organizationCode && workspace.projectCode))
-    ).subscribe(workspace => {
-      this.organizationCode = workspace.organizationCode;
-      this.projectCode = workspace.projectCode;
-      this.getProject();
-      this.originalProjectCode = this.projectCode;
-    });
+  public ngOnInit() {
+    this.subscribeToStore();
   }
 
-  private getProject(): void {
-    this.projectService.getProject(this.organizationCode, this.projectCode)
-      .subscribe(
-        (project: Project) => {
-          this.project = project;
-          this.getNumberOfCollections();
-          this.originalProjectName = this.project.name;
-        },
-        error => {
-          const message = this.i18n({id: 'project.get.fail', value: 'Failed to get project.'});
-          this.notificationService.error(message);
-        }
-      )
-    ;
-  }
+  public ngOnDestroy() {
+    this.subscription.unsubscribe();
 
-  public updateProject(): void {
-    this.projectService.editProject(this.organizationCode, this.projectCode, this.project).subscribe(
-      success => null,
-      error => {
-        const message = this.i18n({id: 'project.update.fail', value: 'Failed to update project.'});
-        this.notificationService.error(message);
-      });
+    this.store.dispatch(new CollectionsAction.Clear());
```

Comment:
Why are collections cleared when you leave project settings page?

Reply:
I will discuss this on tomorrow meeting.